CIP-0062/README.md

```diff
@@ -0,0 +1,220 @@
+
+## **api.submitDelegation(delegation: Delegation): Promise\<SignedDelegationMetadata>**
+
+This endpoint should construct the cbor encoded delegation certificate according to the specs in [CIP-36 Example](https://github.com/Zeegomo/CIPs/blob/472181b9c69feeedae0b5b2db8b42d0cf4eb1a11/CIP-0036/README.md#example).
+
+It should then sign the certificate with the staking key as described in the same example as above.
+
+The implementation of this endpoint can make use of the already existing [CIP-30](https://cips.cardano.org/cips/cip30/) `api.signData` and `api.submitTx` to perform the broadcasting of the transaction containing the metadata. 
+
+Upon submission of the transaction containing the delegation cert as part of metadata, the wallet should store the delegation in its local storage and return an object that contains the delegation cert, the signature and the txhash of the transaction that the certificate was submitted with.
+
+## **`Delegation`**
+
+```
+export interface Delegation {
+    voting_delegation: GovernanceKey[],
+    purpose: Purpose
+}
+```
+
+Defines the structure to be crafted and signed for delegation of voting & their respectively voting power.   Embeds the stake key and reward address from the wallet, and constructs a suitable nonce.
+
+***`voting_delegation`***: List of keys and their voting weight to delegate voting power to.
+
+This should be a call that implicitly cbor encodes the `delegation` object and uses the already existing [CIP-30](https://cips.cardano.org/cips/cip30/) `api.submitTx` to submit the transaction. The resulting transaction hash should be returned.
+
+This should be trigger a request to the wallet to approve the transaction.
+
+### **Returns**
+
+```
+interface SignedDelegationMetadata {
+  certificate: DelegatedCertificate,
+  signature: string, // signature on the CIP-36 delegation certificate
+  txHash: string // of the transaction that submitted the delegation 
+}
+```
+
+#### **DelegatedCertificate**
+```
+interface DelegatedCertificate {
+  delegations: GovernanceKey[] // key delegations,
+  stakingPub: string // staking public key
+  rewardAddress: string // reward address
+  nonce: number //nonce used in the transaction
+  purpose: Purpose
+}
+```
+
+Errors: `APIError`, `TxSendError`
+
+## **Delegation Cert process**
+
+1. **`Get Voting Key`** - use the method **api.getVotingKey** to return a ed25519 32 bytes public key (x value of the point on the curve).
+
+2. **`Collect Voting Keys`** - Collect the keys to delegate voting power to.
+
+3. **`Craft delegation cert`** - Use **api.buildDelegation** to construct the object containing the key array set to delegate voting power to. Each value will express the `weight` of the voting powers given.
+
+4. **`Submit delegation`** - Submit the metadata transaction to the chain using **api.submitDelegation** which implicitly sign and/or can use the already existing **api.submitTx**,  available from [CIP-30](https://cips.cardano.org/cips/cip30/)
```

Comment:
This proposal is missing a rationale section. As far as I can tell, a lot of those operations (like the creation of the metadata itself) can be achieved with client libraries such as the cardano-serialization-lib. Thus, what's the justification for adding them to the wallet connector?

Reply:
(don't say it to me, write it in the proposal!)